### PR TITLE
[6484] - implement new view with timeline, add methods that handle th…

### PR DESCRIPTION
**Pull request by ADubielSTX**
**Story / Bug id**: https://bugzilla.stxnext.pl/show_bug.cgi?id=6484
**Project / App**: Onboarding - Presence Analyzer
**Reference person**: @ADubielSTX
**Description**: Implement new view mentioned in ticket.

**Migrations**:

**Rollback**:

**New imports / dependencies**: 

**Unit Tests**:
New tests in /src/presence_analyzer/tests.py

**What tests do I need to run to validate this change**:

## New or Changed JS and / or HTML Files:
Changed
/src/presence_analyzer/static/mean_time_weekday.html
/src/presence_analyzer/static/presence_weekday.html
Added
/src/presence_analyzer/static/presence_start_end.html

### CODE REVIEW

**Comment by ADubielSTX** [comment]
Gist with PEP8, Pylint, unittest result etc.
https://gist.github.com/ADubielSTX/2c1d63b7477ca4497552#file-6484-pull-request-info-L6

File: src/presence_analyzer/static/presence_start_end.html [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                    loading.hide();
+                });
+                $('#user_id').change(function(){
+                    var selected_user = $("#user_id").val();
+                    var chart_div = $('#chart_div');
+                    if(selected_user) {
+                        loading.show();
+                        chart_div.hide();
+                        $.getJSON("/api/v1/presence_start_end/"+selected_user, function(result) {
+	                        // some dummy data, this should go from API
```
Comment (piasekps) [inline src/presence_analyzer/static/presence_start_end.html]: We don't need that in code. Please remove it

File: src/presence_analyzer/tests.py [hunk 2]
```diff
@@ -88,6 +87,18 @@ def test_presence_weekday_view(self):
...
 
+    def test_presence_start_end_view(self):
+        """
+        Test presence mean time start/end for given user grouped by weekday.
+        """
+        resp = self.client.get('/api/v1/presence_start_end/666')
+        self.assertEqual(resp.status_code, 404)
+        self.assertEqual(resp.content_type, 'text/html')
+
+        resp = self.client.get('/api/v1/presence_start_end/10')
```
Comment (piasekps) [inline src/presence_analyzer/tests.py]: Can we split this into second test, so we have only one case per test?

File: src/presence_analyzer/utils.py [hunk 3]
```diff
@@ -83,6 +83,43 @@ def group_by_weekday(items):
     return result
 
 
+def group_start_end(items):
+    """
+    Groups presence start and end entries by weekday.
+    """
+    result = [
```
Comment (piasekps) [inline src/presence_analyzer/utils.py]: Can you do it more pythonic way using list comprehension?

File: src/presence_analyzer/utils.py [hunk 4]
```diff
@@ -83,6 +83,43 @@ def group_by_weekday(items):
     return result
 
 
+def group_start_end(items):
+    """
+    Groups presence start and end entries by weekday.
```
Comment (piasekps) [inline src/presence_analyzer/utils.py]: In docstring should be also some information about parameter used and what method is return, please add it here and in other methods.

File: src/presence_analyzer/views.py [hunk 5]
```diff
@@ -4,12 +4,13 @@
 """
 
 import calendar
+import logging
 from flask import redirect, abort
 
 from presence_analyzer.main import app
 from presence_analyzer.utils import jsonify, get_data, mean, group_by_weekday
+from presence_analyzer.utils import mean_start_stop
```
Comment (piasekps) [inline src/presence_analyzer/views.py]: first above you have already some methods from utils, so why did you add it here another one?

File: src/presence_analyzer/views.py [hunk 6]
```diff
@@ -4,12 +4,13 @@
 """
 
 import calendar
+import logging
 from flask import redirect, abort
 
 from presence_analyzer.main import app
 from presence_analyzer.utils import jsonify, get_data, mean, group_by_weekday
```
Comment (piasekps) [inline src/presence_analyzer/views.py]: if you will have to use 10 methods will you import them the same way? it's start to be annoying. can you please make it more readable?

File: src/presence_analyzer/static/presence_start_end.html [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+	                        //    ["Tue", new Date(1, 1, 1, 8, 16, 23), new Date(1, 1, 1, 14, 01, 47)],
+	                        //    ["Wed", new Date(1, 1, 1, 7, 00, 06), new Date(1, 1, 1, 13, 24, 12)],
+	                        //    ["Thu", new Date(1, 1, 1, 7, 26, 32), new Date(1, 1, 1, 12, 40, 45)],
+	                        //    ["Fri", new Date(1, 1, 1, 6, 53, 22), new Date(1, 1, 1, 12, 52, 09)]
+	                        //]
+	                        $.each(result, function(index, value) {
+                            	value[1] = parseInterval(value[1]);
+                            	value[2] = parseInterval(value[2]);
+                            });
+                            console.log(result);
```
Comment (piasekps) [inline src/presence_analyzer/static/presence_start_end.html]: do we really need that?

File: src/presence_analyzer/tests.py [hunk 8]
```diff
@@ -139,6 +150,26 @@ def test_group_by_weekday(self):
...
+            utils.group_by_weekday({'first': 123, 'second': '123'})
+
+    def test_mean_start_stop(self):
+        """
+        Test counting starts and ends by weekday.
+        """
+        data = utils.get_data()
+        counted_by_weekday = utils.mean_start_stop(data[10])
+        self.assertIsInstance(counted_by_weekday, list)
+        with self.assertRaises(TypeError):
```
Comment (piasekps) [inline src/presence_analyzer/tests.py]: I guess that could be seperate test

File: src/presence_analyzer/tests.py [hunk 9]
```diff
@@ -88,6 +87,18 @@ def test_presence_weekday_view(self):
         self.assertEqual(resp.status_code, 200)
         self.assertEqual(resp.content_type, 'application/json')
 
+    def test_presence_start_end_view(self):
```
Comment (piasekps) [inline src/presence_analyzer/tests.py]: it would be nice to see if test return correct data, what do you think?

**Comment by piasekps** [comment]
review finished.

File: src/presence_analyzer/static/presence_start_end.html (hunk 1 above)
Comment (ADubielSTX) [inline src/presence_analyzer/static/presence_start_end.html]: Right. Removed.

File: src/presence_analyzer/tests.py (hunk 2 above)
Comment (ADubielSTX) [inline src/presence_analyzer/tests.py]: All (or almost all) tests are divided.

File: src/presence_analyzer/utils.py (hunk 3 above)
Comment (ADubielSTX) [inline src/presence_analyzer/utils.py]: Thanks for that hint. I forgot about comprehensions.

File: src/presence_analyzer/utils.py (hunk 4 above)
Comment (ADubielSTX) [inline src/presence_analyzer/utils.py]: Done.

File: src/presence_analyzer/views.py (hunk 5 above)
Comment (ADubielSTX) [inline src/presence_analyzer/views.py]: I added in new line because previous line would be longer than 80 chars.

File: src/presence_analyzer/views.py (hunk 6 above)
Comment (ADubielSTX) [inline src/presence_analyzer/views.py]: You are right. I just focused on implementing new view and forgot about keeping an eye on code readability by the way.

File: src/presence_analyzer/static/presence_start_end.html (hunk 7 above)
Comment (ADubielSTX) [inline src/presence_analyzer/static/presence_start_end.html]: I just overlooked that (copy/paste from other .html and modified what was needed and not focused on clearing code from lines like that - sorry).

File: src/presence_analyzer/tests.py (hunk 9 above)
Comment (ADubielSTX) [inline src/presence_analyzer/tests.py]: Yes, it would be nice. Tests correct return data added to all view tests.

**Comment by ADubielSTX** [comment]
Additional info (PEP8, tests, pylint)

https://gist.github.com/ADubielSTX/90d8ff3e9e6526513ada

**Comment by piasekps** [comment]
Reveiw finished
